content/docs/guides/author_develop/pwa-amp/amp-in-pwa.md

```diff
@@ -24,6 +24,31 @@ Include Shadow AMP in the head of your page, like so:
 <script async src="https://cdn.ampproject.org/shadow-v0.js"></script>
 [/sourcecode]
 
+### How do you know when Shadow AMP is ready to use?
+
+We recommend you load the Shadow AMP library with the `async` attribute in place. That means, however, that you need to use a certain approach to understand when the library is fully loaded and ready to be used.
+
+The right signal to observe is the availability of the global `AMP` variable, and Shadow AMP uses a “[asynchronous function loading approach](http://mrcoles.com/blog/google-analytics-asynchronous-tracking-how-it-work/)” to help with that. Consider this code:
+
+[sourcecode:javascript]
+(window.AMP = window.AMP || []).push(function(AMP) {
+  // AMP is now available.
+});
+[/sourcecode]
+
+This code will work, and the callback will indeed fire when AMP is available, but why?
+
+This code translates to:
+
+  1. “if window.AMP doesn't exist, create an empty array to take its position”
+  1. "then push a callback function into the array that should be executed when AMP is ready"
```

Comment:
May or may not be worth mentioning that the model directly allows any number of callbacks.